Fixes a couple of problems of [PR #7679]

wwmayer opened the pull request:
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

PaddleStroke commented:
Those ended up being big changes in the end.
I will try to integrate those good practices: 
- Fix CI warnings in a separate commit after creating the PR.
- Use new connect syntax and fix old one on code I work on.

wwmayer commented:
First I only wanted to do the two mentioned points but then I also fixed the other old-style connect, fixed several warnings reported by GH actions and moved all the access of user parameters to a central class.

freecadci commented:
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7744"><img alt="pipeline status" src="https://gitlab.com/freecad/FreeCAD-CI/badges/PR_7744/pipeline.svg"/></a> for feature branch [PR_7744](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7744). Pipeline [689009417](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/689009417) was triggered at [d862dc2](https://github.com/FreeCAD/FreeCAD/pull/7744/commits/d862dc285d530a6791ea424070023c874c52199b). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

donovaly commented:
This PR seems ready to be merged, right? But it has merge conflicts.

wwmayer commented:
I force pushed the rebased branch.

donovaly commented:
Thanks. Merged.
